uint128/uint128_test.go

```diff
@@ -0,0 +1,396 @@
+package uint128
+
+import (
+	"encoding/json"
+	"math/big"
+	"testing"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+)
+
+func TestUint128Bytes(t *testing.T) {
+	tests := []struct {
+		description  string
+		loBits       uint64
+		hiBits       uint64
+		expected_arr []byte
+		expected_str string
+		wantedErr    bool
+	}{
+		{
+			description:  "128-bit #1",
+			loBits:       0x0,
+			hiBits:       0x0,
+			expected_arr: []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+			expected_str: "0x0",
+			wantedErr:    false,
+		},
+		{
+			description:  "128-bit #2",
+			loBits:       0x1,
+			hiBits:       0x0,
+			expected_arr: []byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1},
+			expected_str: "0x1",
+			wantedErr:    false,
+		},
+		{
+			description:  "128-bit #3",
+			loBits:       0xFFFFFFFFFFFFFFFF,
+			hiBits:       0xFFFFFFFFFFFFFFFF,
+			expected_arr: []byte{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255},
+			expected_str: "0xffffffffffffffffffffffffffffffff",
+			wantedErr:    false,
+		},
+		{
+			description:  "128-bit #4",
+			loBits:       0x0123456789ABCDEF,
+			hiBits:       0x123456789ABCDEF0,
+			expected_arr: []byte{18, 52, 86, 120, 154, 188, 222, 240, 1, 35, 69, 103, 137, 171, 205, 239},
+			expected_str: "0x123456789abcdef00123456789abcdef",
+			wantedErr:    false,
+		},
+		{
+			description:  "128-bit #5",
+			loBits:       0xF0D3B8A289C7E5B3,
+			hiBits:       0x1A4B7E9C2D3F5A6E,
+			expected_arr: []byte{26, 75, 126, 156, 45, 63, 90, 110, 240, 211, 184, 162, 137, 199, 229, 179},
+			expected_str: "0x1a4b7e9c2d3f5a6ef0d3b8a289c7e5b3",
+			wantedErr:    false,
+		},
+		{
+			description:  "128-bit #6",
+			loBits:       0x0,
+			hiBits:       0x1A4B7E9C2D3F5A6E,
+			expected_arr: []byte{26, 75, 126, 156, 45, 63, 90, 110, 0, 0, 0, 0, 0, 0, 0, 0},
+			expected_str: "0x1a4b7e9c2d3f5a6e0000000000000000",
+			wantedErr:    false,
+		},
+		{
+			description:  "128-bit #7",
+			loBits:       0x1,
+			hiBits:       0x1,
+			expected_arr: []byte{0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1},
+			expected_str: "0x10000000000000001",
+			wantedErr:    false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			arr := make([]uint64, 2)
+			arr[0] = test.loBits
+			arr[1] = test.hiBits
+			actual, err := NewInt(arr)
+			if err != nil {
+				if test.wantedErr {
+					return
+				}
+				t.Errorf("couldn't marshal []uint64={%v, %v} into *Int", test.loBits, test.hiBits)
+			}
```

Comment:
```suggestion
			actual := Int(arr)
```
We can remove the `NewInt` function; Go does the work fo us here.